internal/controller/ownerref.go

```diff
@@ -73,3 +74,65 @@ func findFinalOwnerRef(ctx context.Context, client crclient.Client, ns string, o
 	Log.V(1).Info("Pods owner found", "Kind", ownerRef.Kind, "Name", ownerRef.Name, "Namespace", ns)
 	return ownerRef, nil
 }
+
+//FindControllerOwner returns OwnerReferent that owns controller process
+//it starts searching from the current pod and then resolves owners recursively
+//until object without owner is not found
+func FindControllerDeployment(ctx context.Context, client crclient.Client) (*v1.Deployment, error) {
+	ns, err := k8sutil.GetOperatorNamespace()
+	if err != nil {
+		return nil, err
+	}
+
+	// Get current Pod the operator is running in
+	pod, err := k8sutil.GetPod(ctx, client, ns)
+	if err != nil {
+		return nil, err
+	}
+	// Get Owner that the Pod belongs to
+	//TODO Take a look, maybe it's a better thing to use
+	//podOwnerRefs := metav1.NewControllerRef(pod, pod.GroupVersionKind())
+	ownerRef := metav1.GetControllerOf(pod)
+	deployment, err := findDeployment(ctx, client, ns, ownerRef)
+	if err != nil {
+		return nil, err
+	}
+	if deployment != nil {
+		return deployment, nil
+	}
+
+	// Default to returning Pod as the Owner
+	return nil, nil
+}
+
+// findFinalOwnerRef tries to locate the final controller/owner based on the owner reference provided.
+func findDeployment(ctx context.Context, client crclient.Client, ns string, ownerRef *metav1.OwnerReference) (*v1.Deployment, error) {
+	if ownerRef == nil {
+		return nil, nil
+	}
+
+	if ownerRef.Kind == "Deployment" {
+		d := &v1.Deployment{}
+		err := client.Get(ctx, types.NamespacedName{Namespace: ns, Name: ownerRef.Name}, d)
+		d.APIVersion = ownerRef.APIVersion
+		d.Kind = ownerRef.Kind
+		if err != nil {
+			return nil, err
+		}
+		return d, nil
+	}
+	obj := &unstructured.Unstructured{}
+	obj.SetAPIVersion(ownerRef.APIVersion)
+	obj.SetKind(ownerRef.Kind)
+	err := client.Get(ctx, types.NamespacedName{Namespace: ns, Name: ownerRef.Name}, obj)
+	if err != nil {
+		return nil, err
+	}
+	newOwnerRef := metav1.GetControllerOf(obj)
+	if newOwnerRef != nil {
+		return findDeployment(ctx, client, ns, newOwnerRef)
+	}
+
+	Log.V(1).Info("Deployment is not found =(", "Kind", ownerRef.Kind, "Name", ownerRef.Name, "Namespace", ns)
+	return nil, nil
```

Comment:
I'd invert the order of these checks:
```suggestion
	newOwnerRef := metav1.GetControllerOf(obj)
	if newOwnerRef == nil {
		Log.V(1).Info("Deployment is not found =(", "Kind", ownerRef.Kind, "Name", ownerRef.Name, "Namespace", ns)
		return nil, nil
	}
	return findDeployment(ctx, client, ns, newOwnerRef)
```